### ISSUE TITLE
How to handle self-referenced relationship

### ISSUE THREAD

**Issue by tsunamilx** [requery/requery#137]
For example, I have an entity Term that has a parent of its own type, so the relationship looks like Term(parent) to many Terms(child).

How can I handle this?

**Comment by npurushe** [comment]
Currently this isn't supported but its definitely a valid case. Basically you should be able to define a relationship inside the entity that refers to itself.

**Comment by tsunamilx** [comment]
I tried with this, but got "Error:Circular Foreign Key reference found between requerydemo.term.Term and requerydemo.term.Term"
```java
@Entity
public class AbstractTerm implements Persistable {

    @Key
    @Generated
    long id;

    String name;

    @ManyToOne
    AbstractTerm parent;

    @OneToMany(mappedBy = "parent")
    List<AbstractTerm> children;
}
```

**Comment by npurushe** [comment]
Yes not supported yet, will be in the next update most likely

**Comment by tsunamilx** [comment]
I tried again with above example, I got "Warning:Circular Foreign Key reference found between ...", was "Error: ...", so I guess this warning can be ignored?

If I remove 
```java
    @ManyToOne
    AbstractTerm parent;
```
No warning anymore.

**Comment by tsunamilx** [comment]
Update:
When I do `term.setParent(parent)`, it works.
When I do `parent.getChildren().add(child)`, it blows up StackOverflowError, looks like a dead loop.

**Comment by npurushe** [comment]
The StackOverflow is fixed by #150 which will be in the next update shortly
